src/executables/util_cs.c

```diff
@@ -3994,6 +3994,827 @@ tde (UTIL_FUNCTION_ARG * arg)
   return EXIT_FAILURE;
 }
 
+/* TODO : move to flashback_cl.c/h */
+
+static int
+get_class_index (int num_class, OID * oidlist, OID classoid)
+{
+  int idx = 0;
+
+  for (idx = 0; idx < num_class; idx++)
+    {
+      if (OID_EQ (&oidlist[idx], &classoid))
+	{
+	  return idx;
+	}
+    }
+
+  return -1;
+}
+
+static int
+check_and_resize_sql_memory (char **sql, int req_size, int *max_sql_size)
+{
+  char *tmp = NULL;
+
+  if (req_size <= *max_sql_size)
+    {
+      return NO_ERROR;
+    }
+
+  while (*max_sql_size < req_size)
+    {
+      *max_sql_size = *max_sql_size * 2;
+    }
+
+  tmp = (char *) realloc (*sql, *max_sql_size);
+
+  if (tmp)
+    {
+      *sql = tmp;
+    }
+  else
+    {
+      util_log_write_errid (MSGCAT_UTIL_GENERIC_NO_MEM);
+      return ER_OUT_OF_VIRTUAL_MEMORY;
+    }
+
+  return NO_ERROR;
+}
+
+#define IS_QOUTES_NEEDED(type)   (type == DB_TYPE_STRING || \
+                                  type == DB_TYPE_TIME || \
+                                  type == DB_TYPE_TIMESTAMP || \
+                                  type == DB_TYPE_TIMESTAMPTZ || \
+                                  type == DB_TYPE_TIMESTAMPLTZ || \
+                                  type == DB_TYPE_DATE || \
+                                  type == DB_TYPE_DATETIME || \
+                                  type == DB_TYPE_DATETIMETZ || \
+                                  type == DB_TYPE_DATETIMELTZ || \
+                                  type == DB_TYPE_CHAR)
+
+typedef enum
+{
+  PACK_INT = 0,
+  PACK_INT64 = 1,
+  PACK_FLOAT = 2,
+  PACK_DOUBLE = 3,
+  PACK_SHORT = 4,
+  PACK_STRING = 7
+} PACK_FUNC_TYPE;
+
+static int
+process_column_data (char **data, char **sql, int *max_sql_size, DB_TYPE type)
+{
+  char *ptr = *data;
+
+  int error = NO_ERROR;
+
+  int sql_length = strlen (*sql);
+
+  PACK_FUNC_TYPE func_type;
+
+  int i_data;
+  double d_data;
+  float f_data;
+  short sh_data;
+  char *s_data;
+  INT64 b_data;
+
+  int max_decimal_digits_len = 20;
```

Comment:
Could you tell me what does it mean?

Reply:
Simply, it is maximum length of the converted string when converting a decimal number to a string.
INT64 presents 9223372036854775807, and this consist of 20 char.

Reply:
Thank you for the answer. It would be better to leave a comment for it and make it a constexpr.

Reply:
Thanks, I will change it to constexpr, and leave a comment